### PR TITLE
fix: failed to respond to the plugin6.(*GRPCProvider).ApplyResourceCh…

### PR DESCRIPTION
**Pull request by hugomf**
Fix for  "failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call #15"

please take a look I didn't tested in previous versions where the memory flag is like "mem" but i tried to add scenarios to check previous versions.

this change needs to be implemented in the terraform provider project, not sure if someone can help me to do that?
if you don't have much time I will try to do it tomorrow, but I guess this change needs to be approved first

### CODE REVIEW

File: multipass/version_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+	for _, tc := range testCases {
+		got := CompareVersion(tc.prevVersion, tc.mostRecentVersion)
+		if got != tc.expectedResult {
+			t.Errorf("Comparison failed for versions %s and %s. Expected: %d, Got: %d",
+				tc.prevVersion, tc.mostRecentVersion, tc.expectedResult, got)
+		}
+	}
+
+}
```
Comment (larstobi) [inline multipass/version_test.go]:
```suggestion
}

```

File: multipass/version.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	// Compare versions
+	currentversion, err := GetVersion()
+	if err != nil {
+		currentversion = "0.05"
+	}
+	result := CompareVersion(currentversion, v2)
+
+	return result;
+
+}
```
Comment (larstobi) [inline multipass/version.go]:
```suggestion
}

```

File: multipass/multipass.go [hunk 3]
```diff
@@ -12,3 +12,7 @@ type Instance struct {
 	MemoryTotal string
 	Load        string
 }
+
+const (
+	LONG_MEMORY_VERSION = "v1.11.0.rc"
+)
```
Comment (larstobi) [inline multipass/multipass.go]:
```suggestion
)

```

File: multipass/launch_test.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	_, err := LaunchV2(&LaunchReqV2{
+		CPUS: "2",
+		Memory: "2GiB",
+		Name: "instanceName", 
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+}
```
Comment (larstobi) [inline multipass/launch_test.go]:
```suggestion
}

```

**Comment by larstobi** [comment]
@hugomf Thanks! 

I had a look at the CLI options, and it looks like the short option `-m` is available in all versions.  Perhaps it would make sense to use that for all versions, which would give a simpler solution?

File: multipass/launch_test.go [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+	_, err := LaunchV2(&LaunchReqV2{
+		CPUS: "2",
+		Memory: "2GiB",
+		Name: "instanceName", 
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
```
Comment (larstobi) [inline multipass/launch_test.go:33]: We need to stop the instance too.

Comment (hugomf) [reply multipass/launch_test.go:33]: ok, I will stop the instance and I’ll double check about the -m option

**Comment by hugomf** [comment]
removed version check and use -m option

**Comment by hugomf** [comment]
I have finished my changes, please take a look:

1. Removed version check
2. -m option update

**Review by larstobi — APPROVED** [review]
Excellent! Thanks!

File: multipass/launch.go [hunk 6]
```diff
@@ -60,7 +61,7 @@ func LaunchV2(launchReqV2 *LaunchReqV2) (*Instance, error) {
 	}
 
 	if launchReqV2.Memory != "" {
-		args = append(args, "--mem", launchReqV2.Memory)
+		args = append(args, "--m", launchReqV2.Memory)
```
Comment (larstobi) [inline multipass/launch.go]:
```suggestion
		args = append(args, "-m", launchReqV2.Memory)
```

**Pull request merged**